app/modules/tasks/models/task.js

```diff
@@ -21,21 +21,24 @@ export default Backbone.Model.extend({
     })
   },
 
-  grabAuthor (name) {
+  grabAuthor (name, isShownFromId) {
     const uri = this.get(`${name}Uri`)
     return this.reqGrab('get:entity:model', uri, name)
     .then(model => {
       const resolvedUri = model.get('uri')
       if (resolvedUri !== uri) {
-        const context = { task: this.id, oldUri: uri, newUri: resolvedUri }
-        throw error_.new(`${name} uri is obsolete`, 500, context)
+        if (isShownFromId) {
+          const context = { task: this.id, oldUri: uri, newUri: resolvedUri }
+          throw error_.new(`${name} uri is obsolete`, 500, context)
+        }
+        return this.showNextTask()
```

Comment:
This is view logic (as hinted by the `show` terminology) and shouldn't be in the model. It might even have been the initial intention has `this.showNextTask` is not defined on the model but on `tasks_layout`, thus the error `this.showNextTask is not a function`

To implement this behavior from the view, I would suggest to catch the error thrown by the model from the view, and call `this.showNextTask` if the error matches that known exception

Reply:
definetly